packages/paste-website/src/components/icons-list/IconsList.tsx

```diff
@@ -0,0 +1,145 @@
+  };
+}, {});
+
+const getGroupedList = (icons: IconsListProps['icons']): GroupedList =>
+  icons.reduce(
+    (prev: GroupedList, current): GroupedList => {
+      const Icon = {...current, component: IconComponents[current.name]};
+      if (current.name.startsWith('Product')) {
+        return {
+          products: [...prev.products, Icon],
+          ui: [...prev.ui],
+        };
+      }
+      return {
+        products: [...prev.products],
+        ui: [...prev.ui, Icon],
+      };
+    },
+    {products: [], ui: []}
+  );
+
+const getFirstIcon = (iconsList: GroupedList): IconObject | null => {
+  if (iconsList.ui.length !== 0) {
+    return iconsList.ui[0];
+  }
+  if (iconsList.products.length !== 0) {
+    return iconsList.products[0];
+  }
+  return null;
+};
+
+const IconTileWrapper = styled(Box)({
+  width: '33.33%',
+});
+
+const IconColumn = styled(Box)({
+  width: '41.66666666666667%',
+});
+
+const IconsList: React.FC<IconsListProps> = () => {
+  const {isPasteTheme} = React.useContext(SiteBodyContext);
+  const filterID = useUID();
+  const iconKeySeed = useUIDSeed();
+  const groupedIconList = getGroupedList(iconsJson);
+  const [filterString, setFilterString] = React.useState('');
+  const [iconsList, setIconsList] = React.useState(groupedIconList);
+  const [selectedIcon, setSelectedIcon] = React.useState(getFirstIcon(iconsList));
+  return (
+    <Box maxWidth="size120">
+      <Grid gutter="space60">
+        <Column span={7}>
+          <Box marginBottom="space70">
+            <FormLabel htmlFor={filterID}>Filter Icons</FormLabel>
+            <FormInput
+              id={filterID}
+              type="text"
+              value={filterString}
+              placeholder="Filter by name..."
+              onChange={e => {
+                const filteredList = iconsJson.filter(({name}: IconObject) =>
+                  name.toLowerCase().includes(e.currentTarget.value.toLowerCase())
+                );
+                const filteredGroupedList = getGroupedList(filteredList);
+                setIconsList(filteredGroupedList);
+                setSelectedIcon(getFirstIcon(filteredGroupedList));
+                setFilterString(e.currentTarget.value);
+              }}
+            />
+          </Box>
+        </Column>
+        <Column span={7}>
+          {iconsList.ui.length > 0 && (
+            <>
+              <Heading as="h2" variant="heading20">
+                UI Icons
+              </Heading>
+              <Box marginX="spaceNegative40" marginBottom="space70" display="flex" flexWrap="wrap">
+                {iconsList.ui.map((icon, index) => {
+                  if (icon.name !== 'index') {
+                    return (
+                      <IconTileWrapper paddingX="space40" marginBottom="space60" key={iconKeySeed(`icon-${index}`)}>
+                        <IconTile icon={icon} onClick={() => setSelectedIcon(icon)}>
+                          <icon.component decorative size="sizeIcon70" />
```

Comment:
nit: It's odd seeing react components not start with a capital letter, so can we do
```
{iconsList.ui.map(({name, component: Component}, index) => {
....
  <Component decorative .../>
```